File: LeetCode/Easy/0342-power-of-four/0342-power-of-four.cs
public class Solution {
    public bool IsPowerOfFour(int n)
        => n > 0 && (n & (n - 1)) == 0 && (n & 0x55555555) != 0;
}

// classic solution
// public class Solution {
//     public bool IsPowerOfFour(int n) {
//         if (n <= 0) return false;
//         while (n % 4 == 0) n /= 4;
//         return n == 1;
//     }
// }

File: LeetCode/Easy/0342-power-of-four/0342-power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:    
        if n <= 0:
            return False
        
        # with modulo 3
        return n & (n - 1) == 0 and n % 3 == 1
        
        # with bit mask
        return n & (n - 1) == 0 and n & 0x55555555 != 0